plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/p5edges/orthogonal/direction/NorthToSouthRoutingStrategy.java

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+ * Copyright (c) 2010, 2020 Kiel University and others.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.elk.alg.layered.p5edges.orthogonal.direction;
+
+import org.eclipse.elk.alg.layered.graph.LEdge;
+import org.eclipse.elk.alg.layered.graph.LPort;
+import org.eclipse.elk.alg.layered.p5edges.orthogonal.HyperEdgeSegment;
+import org.eclipse.elk.alg.layered.p5edges.orthogonal.OrthogonalRoutingGenerator;
+import org.eclipse.elk.core.math.KVector;
+import org.eclipse.elk.core.options.PortSide;
+
+/**
+ * Routing strategy for routing layers from north to south.
+ */
+class NorthToSouthRoutingStrategy extends BaseRoutingDirectionStrategy {
+
+    @Override
+    public double getPortPositionOnHyperNode(final LPort port) {
+        return port.getNode().getPosition().x + port.getPosition().x + port.getAnchor().x;
+    }
+
+    @Override
+    public PortSide getSourcePortSide() {
+        return PortSide.SOUTH;
+    }
+
+    @Override
+    public PortSide getTargetPortSide() {
+        return PortSide.NORTH;
+    }
+
+    @Override
+    public void calculateBendPoints(final HyperEdgeSegment segment, final double startPos, final double edgeSpacing) {
```

Comment:
What is the difference of this `calculateBendPoints` and the one in `SouthToNorthRoutingStrategy` and `WestToEastRoutingStrategy`? It looks like lots of duplicate code.

Reply:
As the names suggest, they are exchangable strategies that will be used depending on the routing direction.

Reply:
@soerendomroes is right in that the code is very similar. However, I fear that turning it into a single implementation would be way harder to understand since it would be riddled either with `if` statements or with calls to strategies that determine whether to use an x or a y coordinate at a given place.